File: recursion.py
# recursion.py
def factorial(n):
    if n <= 1:
        return 1
    else:
        return n * factorial(n - 1)

input_value = input("Enter an integer: ")
try:
    param = int(input_value)
except ValueError:
    print("Invalid input")
else:
    print(factorial(param))

File: recursion.php
// recursion.php
function factorial($param) {
    if ($param <= 1) {
        return 1;
    } else {
        return $param * factorial($param - 1);
    }
}

echo "Enter an integer: ";
$input = trim(fgets(STDIN));
$param = intval($input);

if ($param == 0 && $input !== "0") {
    echo "Invalid input\n";
} else {
    echo factorial($param) . "\n";
}
